state/src/state.rs

```diff
@@ -0,0 +1,73 @@
+use gear_lib::non_fungible_token::state::{NFTQuery, NFTQueryReply};
+use gear_lib::non_fungible_token::token::{TokenId, TokenMetadata};
+use gmeta::{metawasm, Metadata};
+use gstd::prelude::*;
+use onchain_nft_io::*;
+
+#[metawasm]
+pub trait Metawasm {
+    type State = <ContractMetadata as Metadata>::State;
+
+    fn token_uri(token_id: TokenId, state: Self::State) -> Option<Vec<u8>> {
+        let metadata = if let Some((_id, metadata)) = state
+            .token
+            .token_metadata_by_id
+            .iter()
+            .find(|(id, _)| token_id.eq(id))
+        {
+            metadata.clone().unwrap_or_default()
+        } else {
+            TokenMetadata::default()
+        };
+        // construct media
+        let mut content: Vec<String> = Vec::new();
+        // check if exists
+
+        if let Some((_id, nft)) = state.nfts.iter().find(|(id, _)| token_id.eq(id)) {
+            for (i, layer_item_id) in nft.iter().enumerate() {
+                if let Some((_id, layer_content)) =
+                    state.layers.iter().find(|(id, _)| (i as u128).eq(id))
+                {
+                    let s = layer_content
+                        .iter()
+                        .nth(*layer_item_id as usize)
+                        .expect("No such layer item");
+                    content.push(s.clone());
+                    return Some(TokenURI { metadata, content }.encode());
```

Comment:
```suggestion
```
Returning on the first iteration? In the source function this statement was at the end of the block.